### PR TITLE
Unite underlying log libraries into Zap

### LINKED ISSUE #883: Unite underlying log libraries into Zap
Our project has chosen to use Zap for structured logging. Some of our dependencies (Mako, client-go) use other logging libraries: glog, klog. Those two don't coexist well and we shouldn't have several knobs to control logging.

Istio team had this same issue and has a very neat solution of writing API wrapper with calls to Zap, and updating go.mod to redirect imports for github.com/golang/glog and k8s.io/klog to the wrappers. We can do the same thing with Go dep and reuse Istio's wrappers.

Then all logging can be controlled with one package. This needs to be done in all repos.

Istio wrappers:
https://github.com/istio/klog/
https://github.com/istio/glog/

### PR DESCRIPTION
**Pull request by coryrc**
Fixes #883.

Currently uses my forks of glog/klog proxies until knative-community repos can be setup for them.

### CODE REVIEW

File: test/logging/example_tlogger_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	t, cancel := logging.NewTLogger(legacy)
+	defer cancel()
+
+	// For the most part, you can pretend t is really a *testing.T
+	// But you get better results by treating it as a leveled logger
+	// with the same semantics as Infow() in Zap (keys & values alternating after the main argument).
+	// Logging is leveled from 0-10; see https://github.com/go-logr/logr#how-do-i-choose-my-v-levels
+	// In our tests currently, levels are sort-of used as follows:
+	// 1: Describe broadly what the test is doing
+	// 2: What specific action is occuring
```
Comment (mattmoor) [inline test/logging/example_tlogger_test.go]: Found potential misspelling(s):
```suggestion
	// 2: What specific action is occurring
```

**Comment by coryrc** [comment]
Please consider only the "Use klog/glog proxies" commit

**Comment by coryrc** [comment]
Running `./hack/update-codegen.sh` doesn't do anything on my computer, so I don't understand why it's trying to add:

```
+    "k8s.io/client-go/discovery/fake",
...
+    "k8s.io/client-go/util/flowcontrol",
```

If I take them out manually, ./hack/update-codegen.sh adds them back in.

**Comment by vagababov** [comment]
Do we still pursue this?

**Comment by coryrc** [comment]
This is trivial to repeat (and needs to use Go modules instead) so closing for now until the knative-sandbox repos are created for glog and klog.

**Pull request closed without merging**